zerver/migrations/0691_backfill_message_is_channel_message.py

```diff
@@ -0,0 +1,39 @@
+from django.db import migrations, transaction
+from django.db.backends.base.schema import BaseDatabaseSchemaEditor
+from django.db.migrations.state import StateApps
+from django.db.models import OuterRef, Q, Subquery
+
+
+def update_is_channel_message(apps: StateApps, schema_editor: BaseDatabaseSchemaEditor) -> None:
+    Realm = apps.get_model("zerver", "Realm")
+    Recipient = apps.get_model("zerver", "Recipient")
+    Message = apps.get_model("zerver", "Message")
+    ArchivedMessage = apps.get_model("zerver", "ArchivedMessage")
+
+    for realm in Realm.objects.all():
+        with transaction.atomic(durable=True):
+            Message.objects.filter(realm=realm, is_channel_message__isnull=True).alias(
+                # We do this as a subquery due to https://code.djangoproject.com/ticket/14104
+                recipient_type=Subquery(
+                    Recipient.objects.filter(id=OuterRef("recipient_id")).values("type")[:1]
+                )
+            ).update(is_channel_message=Q(recipient_type=2))
+            ArchivedMessage.objects.filter(realm=realm, is_channel_message__isnull=True).alias(
+                recipient_type=Subquery(
+                    Recipient.objects.filter(id=OuterRef("recipient_id")).values("type")[:1]
+                )
+            ).update(is_channel_message=Q(recipient_type=2))
```

Comment:
We might need to batch this by message ID rather than doing per-realm. Individual realms can have 10M+ messages, and I'd expect that to be a very long query that might conflict with new messages being added.